components/global/hooks.tsx

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {MutableRefObject, useEffect} from 'react';
+import {useSelector} from 'react-redux';
+import {useLocation} from 'react-router';
+
+import {GlobalState} from 'types/store';
+import {ProductComponent} from 'types/store/plugins';
+import {getBasePath} from 'utils/url';
+
+const selectProducts = (state: GlobalState) => state.plugins.components.Product;
+
+export const useProducts = (): ProductComponent[] | undefined => {
+    return useSelector<GlobalState, ProductComponent[]>(selectProducts);
+};
+
+/**
+ * Hook that alerts clicks outside of the passed ref.
+ */
+export function useClickOutsideRef(ref: MutableRefObject<HTMLElement | null>, handler: () => void): void {
+    useEffect(() => {
+        function onMouseDown(event: MouseEvent) {
+            const target = event.target as any;
+            if (ref.current && target instanceof Node && !ref.current.contains(target)) {
+                handler();
+            }
+        }
+
+        // Bind the event listener
+        document.addEventListener('mousedown', onMouseDown);
+        return () => {
+            // Unbind the event listener on clean up
+            document.removeEventListener('mousedown', onMouseDown);
+        };
+    }, [ref, handler]);
+}
```

Comment:
And I took it from playbooks. Its a clearly useful hook.

Reply:
We may wish to change this eventually to actually use an invisible div behind the element when we want the click to close behaviour. This is nice and elegant, but there's a bunch of weirdness that can come from being able to interact with things behind a menu like accidentally clicking on stuff or scrolling the object that contains the menu. That's not something that needs to change now, but I feel like it might apply to compass-components more